### PR TITLE
Clear cache button and log out account

### LINKED ISSUE #301: Clean local cache button
My use case:

1. iPhone full of pictures
2. Backed up to owncloud using iOS client
3. Removed pictures form the phone
4. Free space is the same as before (owncloud local cache?) 
5. Reinstalled the app to drop owncloud cache fixed free space issue.

It would be useful to have "Clean local cache" button.

Thanks

### LINKED ISSUE #640: Multiple users and forced log off
--> I work for a small school district and we are currently testing owncloud as a way for students to access their network drives from any device.  We have a large number of shared ipads in the schools and were hoping to use the app there but I am a unable to find a way to have them log out of the app.

Ideally, it would log them out after a certain # of minutes or when the device locks.  Even an easy to find log out button would be useful.  I can't figure out how to log mine out all all without going into settings.

### LINKED ISSUE #657: [Clear cache&Remove account] As an iOS user I want to be able to delete the downloaded data on my device so that I don't run out of space
AC:
- Add an button to delete all the downloaded data on an account -> the selected account #301
- This user story includes the change of place of the option to log out. #640
- UI: #640 (comment)

IMPORTANT:
- Favorite offline synced data won't be removed
- Check single account vs multiaccount

TASKS:
- [x] [GIT] Create branch https://github.com/owncloud/ios/tree/clear_cache_button @javiergonzper 
- [X] [DEV] Add direct clear cache button in settings view @javiergonzper 
- [X] [DEV] Create delete all local files downloaded except the favourites @javiergonzper 
- [X] [DEV] Update settings view in order to use old look but remain the refactor per multiaccount, options per account will be set in other menu @nasli
- [X]  [DEV] Add dot button instead of current info icon in account rows in Settings view @nasli
- [X] [DEV] Add new menu view with options "Edit Account", "Clear Cache", "Log out"  @nasli
- [X] [DEV] Add logic edit account in menu option @nasli
- [X] [DEV] Add logic log out in menu option @nasli
- [X] [DEV] Updated view in order to use also disclosure instead of dots button [Won't Fix] @nasli We use the event of tap in the row to select active account, we would need to use DetailDisclosureButton instead of the disclosureIndicator and now looks like
detaildisclosurebutton. For the moment we consider better UX use the three dot button
- [X] [DEV] Check Title to correct fit with longer accounts names  @nasli
- [X] [DEV] Check constraints username account cell @nasli
- [X] [DEV] Add logic clear cache new button menu @nasli 
- [X] [DEV] Update title menu with "@" between username and account url @nasli
- [X] [BUG] Crash after switch between accounts @nasli [cannot reproduce]
- [X] [DEV] Update wording "Edit Credentials" @nasli
- [X] [DEV] Update wording "Remove account" @nasli
- [X] [DEV] Define single account UI. Show account also account section @nasli
- [X] [QA] Create Test plan: https://github.com/owncloud/QA/issues/235 @jesmrec 
- [X] [QA] Validate Test plan @jesmrec
- [X] [GIT] Merge branch "clear_cache_button" @mRoRo 

BUGS:
- [X] Option "Edit Credentials" does not work in iPad https://github.com/owncloud/ios/issues/672 @nasli
- [X] Clearing cache with a spinner https://github.com/owncloud/ios/issues/673 @mRoRo
- [X] Menu chart displaced in iPad https://github.com/owncloud/ios/issues/675 @nasli

PR:https://github.com/owncloud/ios/pull/669

### PR DESCRIPTION
**Pull request by nasli**
This closes #657 , closes #301 , and closes #640

### CODE REVIEW

File: Owncloud iOs Client/Utils/OCLoadingSpinner.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (mRoRo) [inline Owncloud iOs Client/Utils/OCLoadingSpinner.h]: Should add header

File: Owncloud iOs Client/Utils/OCLoadingSpinner.m [hunk 2]
```diff
@@ -0,0 +1,90 @@
+//
```
Comment (mRoRo) [inline Owncloud iOs Client/Utils/OCLoadingSpinner.m]: Should be added the header too

File: Owncloud iOs Client/Utils/DeleteUtils.h [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//
```
Comment (mRoRo) [inline Owncloud iOs Client/Utils/DeleteUtils.h]: The OC header should be added

File: Owncloud iOs Client/Utils/DeleteUtils.m [hunk 4]
```diff
@@ -0,0 +1,42 @@
+//
```
Comment (mRoRo) [inline Owncloud iOs Client/Utils/DeleteUtils.m]: The OC header should be added

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 5]
```diff
@@ -319,14 +325,7 @@ -(void)goImprint {
 // Asks the data source to return the number of sections in the table view.
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Great refactor of this class!

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 6]
```diff
@@ -828,11 +752,57 @@ -(UITableViewCell *) getSectionAppInstantUpload:(UITableViewCell *) cell byRow:(
     return cell;
 }
 
+- (AccountCell *) getActiveAccountCell {
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: This method seems unused

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 7]
```diff
@@ -901,6 +875,26 @@ - (UITableViewCell *) getSectionAddAccountButton:(UITableViewCell *) cell byRow:
     return addAccountCell;
 }
 
+- (UITableViewCell *) getSectionClearCache:(UITableViewCell *) cell byRow:(NSInteger) row {
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: This method seems unused

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 8]
```diff
@@ -995,87 +959,14 @@ - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEd
 {
     DLog(@"DELETE!!! %ld", (long)indexPath.row);
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Probably this comment is not necessary

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 9]
```diff
@@ -995,87 +959,14 @@ - (void)tableView:(UITableView *)tableView commitEditingStyle:(UITableViewCellEd
...
-            //[self performSelectorInBackground:@selector(cancelAllDownloads) withObject:nil];
-            app.uploadArray=[[NSMutableArray alloc]init];
-            [app updateRecents];
-            [app restartAppAfterDeleteAllAccounts];
-        }
-        
+         UserDto *selectedUser = (UserDto *)[self.listUsers objectAtIndex:indexPath.row];
+        [self dicSelectLogOutAccount:selectedUser];
         
     } else if (editingStyle == UITableViewCellEditingStyleInsert) {
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Please check if the else if structure qill be used in the future

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 10]
```diff
@@ -2061,73 +1983,156 @@ -(void) changedLocation{
...
+    [self performSelectorInBackground:@selector(cancelAllDownloads) withObject:nil];
+    
+    [[ManageThumbnails sharedManager] deleteThumbnailCacheFolderOfUserId: user.idUser];
+    
+    //Delete the tables of this user
+    [ManageUsersDB removeUserAndDataByIdUser: user.idUser];
+    
+    [self performSelectorInBackground:@selector(cancelAndRemoveFromTabRecentsAllInfoByUser:) withObject:user];
+    
+    //Delete files os user in the system
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: "os user" -> "of user"

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m [hunk 11]
```diff
@@ -2061,73 +1983,156 @@ -(void) changedLocation{
...
+    if([self.listUsers count] > 0) {
+        [self.settingsTableView performSelectorOnMainThread:@selector(reloadData) withObject:nil waitUntilDone:NO];
+    } else {
+        
+        self.settingsTableView.editing = NO;
+        
+        AppDelegate *app = (AppDelegate *)[[UIApplication sharedApplication]delegate];
+        
+        [self cancelAllDownloads];
+        //[self performSelectorInBackground:@selector(cancelAllDownloads) withObject:nil];
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Please delete the comment

File: Owncloud iOs Client/Utils/OCLoadingSpinner.h (hunk 1 above)
Comment (nasli) [inline Owncloud iOs Client/Utils/OCLoadingSpinner.h]: 👍 done

File: Owncloud iOs Client/Utils/OCLoadingSpinner.m (hunk 2 above)
Comment (nasli) [inline Owncloud iOs Client/Utils/OCLoadingSpinner.m]: 👍 done

File: Owncloud iOs Client/Utils/DeleteUtils.h (hunk 3 above)
Comment (nasli) [inline Owncloud iOs Client/Utils/DeleteUtils.h]: 👍 done

File: Owncloud iOs Client/Utils/DeleteUtils.m (hunk 4 above)
Comment (nasli) [inline Owncloud iOs Client/Utils/DeleteUtils.m]: 👍 done

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 7 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Yes, now it is on the action sheet this action 😉

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 8 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: 👍 removed

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 9 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: Removed

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 10 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: 👍

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 11 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: 👍

File: Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m (hunk 5 above)
Comment (nasli) [inline Owncloud iOs Client/Tabs/SettingTab/SettingsViewController.m]: 🎉  🎉  🎉  🎉  🎉

**Pull request merged**